src/server.h

```diff
@@ -298,12 +315,14 @@ class Server : public con::PeerHandler, public MapEventReceiver,
 	NodeDefManager* getWritableNodeDefManager();
 	IWritableCraftDefManager* getWritableCraftDefManager();
 
+	// Not under envlock
```

Comment:
This is the only line I didn't understand. Everything else looks good enough Code-wise besides some minor things I mentioned above (///). Design is reasonable, and agree the simplicity isn't perfect, but plenty good for our purposes for now (and probably for a long time).

Reply:
The env mutex does not need to be locked to use the functions that follow the comment, or what do you mean?